File: packages/espresso/ios/Classes/E2EPlugin.m
#import "espressoPlugin.h"

@implementation espressoPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel =
      [FlutterMethodChannel methodChannelWithName:@"plugins.flutter.dev/espresso"
                                  binaryMessenger:[registrar messenger]];
  espressoPlugin* instance = [[espressoPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"allTestsFinished" isEqualToString:call.method]) {
    result(nil);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: packages/e2e/e2e_macos/macos/Classes/E2EPlugin.swift
import FlutterMacOS

public class E2EPlugin: NSObject, FlutterPlugin {

  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
      name: "plugins.flutter.io/e2e",
      binaryMessenger: registrar.messenger)

    let instance = E2EPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "allTestsFinished":
      result(nil)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
